File: MoodprintAndroid/app/src/main/java/com/moodprint/app/domain/MoodDomain.kt
package com.moodprint.app.domain

enum class MoodEmotion(val label: String) {
    ANXIOUS("불안"),
    LETHARGIC("무기력"),
    UPSET("속상함"),
    ANGRY("화남"),
    LONELY("외로움"),
    COMPLICATED("복잡함"),
    TIRED("지침"),
    FRUSTRATED("답답함"),
    SAD("슬픔"),
}

enum class MoodEnergy(val label: String) {
    LOW("낮음"),
    MEDIUM("보통"),
    HIGH("높음"),
}

enum class MoodChange(val label: String, val recommendationWeight: Double) {
    HARDER("더 힘들어요", -0.25),
    SAME("비슷해요", 0.0),
    BETTER("조금 나아졌어요", 0.2),
    MUCH_BETTER("많이 나아졌어요", 0.35),
}

enum class RecoveryActionCategory(val label: String) {
    SENSORY("감각 행동"),
    MOVEMENT("움직임"),
    REST("휴식"),
    EXPRESSION("표현"),
    REFLECTION("회상"),
    ENVIRONMENT("환경"),
}

data class RecoveryAction(
    /** iOS SeedCatalog와 공유하는 변경되지 않는 UUID 문자열이다. */
    val id: String,
    val title: String,
    val instruction: String,
    val durationSeconds: Int,
    val category: RecoveryActionCategory,
    val symbolName: String,
    val supportedEmotions: Set<MoodEmotion>,
    val supportedEnergies: Set<MoodEnergy>,
    val catalogOrder: Int,
    val detailPrompt: String? = null,
    val detailPlaceholder: String? = null,
)

data class ActionHistory(
    val actionId: String,
    val change: MoodChange?,
    val completedAtEpochMillis: Long = 0,
)

data class ActionRecommendation(
    val action: RecoveryAction,
    val score: Double,
    val reason: String,
)

File: Moodprint/Core/Domain/MoodDomain.swift
import Foundation

enum MoodEmotion: String, CaseIterable, Identifiable, Codable, Sendable {
    case anxious = "불안"
    case lethargic = "무기력"
    case upset = "속상함"
    case angry = "화남"
    case lonely = "외로움"
    case complicated = "복잡함"

    var id: String { rawValue }
}

enum MoodEnergy: String, CaseIterable, Identifiable, Codable, Sendable {
    case low = "낮음"
    case medium = "보통"
    case high = "높음"

    var id: String { rawValue }
}

enum MoodChange: String, CaseIterable, Identifiable, Codable, Sendable {
    case harder = "더 힘들어요"
    case same = "비슷해요"
    case better = "조금 나아졌어요"
    case muchBetter = "많이 나아졌어요"

    var id: String { rawValue }

    var recommendationWeight: Double {
        switch self {
        case .harder: -0.25
        case .same: 0
        case .better: 0.2
        case .muchBetter: 0.35
        }
    }
}

enum RecoveryActionCategory: String, CaseIterable, Codable, Sendable {
    case sensory = "감각 행동"
    case movement = "움직임"
    case rest = "휴식"
    case expression = "표현"
    case reflection = "회상"
    case environment = "환경"
}

struct ActionRecommendation: Identifiable {
    let action: RecoveryActionRecord
    let score: Double
    let reason: String

    var id: UUID { action.id }
}

struct RewardOutcome: Equatable, Sendable {
    let experienceAwarded: Int
    let fragmentsAwarded: Int
    let wasAlreadyApplied: Bool

    static let alreadyApplied = RewardOutcome(
        experienceAwarded: 0,
        fragmentsAwarded: 0,
        wasAlreadyApplied: true
    )
}
